pdns/dnsdistdist/docs/reference/config.rst

```diff
@@ -1596,6 +1596,28 @@ Other functions
   If this function exists, it is called every second to do regular tasks.
   This can be used for e.g. :doc:`Dynamic Blocks <../guides/dynblocks>`.
 
+.. function:: threadmessage(cmd, dict)
+
+  .. versionadded:: 1.7.0
+
+  This function, if it exists, is called when a separate thread (made with :func:`newThread`) calls :func:`submitToMainThread`.
+
+.. function:: newThread(code)
+
+  .. versionadded:: 1.7.0
+
+  Spawns a separate thread running the supplied code.
+  Code is supplied as a string, not as a function object.
+
+.. function:: submitToMainThread(cmd, dict)
+
+  .. versionadded:: 1.7.0
+
+  When called from a separate thread (made with :func:`newThread`), submits data to the main thread by calling :func:`threadmessage` in it.
```

Comment:
What happens if called _not_ from a separate thread?

Reply:
`attempt to call a nil value (global 'submitToMainThread')`

Reply:
Better to be normative here: `submitToMainThread must be called from a non-main thread`. That leaves the other case as undefined behaviour explicitly. Otherwise I'm expecting the sentence to continue with: if called from the main thread...

Reply:
amended!